.github/workflows/bootstrapping_ci.yml

```diff
@@ -60,8 +60,15 @@ jobs:
           ls -la v_from_vc2 v_from_vc_produced_native_v2
       - name: Test `v up`
         run: |
-          ./v cmd/tools/oldv.v
-          ./cmd/tools/oldv -v HEAD^^^
-          cd ~/.cache/oldv/v_at_HEAD___/
+          # Derive commit sha from an older successful fast workflow on master that was able to build V.
+          last_good_commit=$(curl -L \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            "https://api.github.com/repos/vlang/v/actions/workflows/18477644/runs?branch=master&status=success&per_page=2" \
+            | jq -r '.workflow_runs[1].head_sha')
+          echo "last_good_commit=$last_good_commit"
+          ./v run cmd/tools/oldv.v -v $last_good_commit
+          cd ~/.cache/oldv/v_at_$last_good_commit
```

Comment:
No, the purpose is not to check that `v up` from a specific commit works, and even if it was, just hardcoding the sha of the known good commit would have been much much simpler, than that curl thing.

The purpose of this task is to test `v up` for recent commits. That way, it will check more things, preferably without needing more updates.

Note, that I do agree that it would be best to test from a good commit, where V could be at least built (to avoid delayed failures like after the recent merge conflict).

However, imho a better tactic for it, to avoid hard coding a known good sha (which will need constant updates to keep it recent), would be to start from say 3 commits ago, then if that does not work, try from 4 commits ago, then 5 etc, up to say 10 commits ago, logging everything.

Reply:
btw, it was not obvious to me, that the `18477644` workflow id is for the `Path Testing CI`, i.e. `paths_ci.yml`.

Is there a convenient way, to see it directly on the github site, without using `gh workflow list` and searching in it, or `gh workflow view 18477644` ?

Reply:
I was wrong - the curl thing does retrieve a known good commit dynamically, just in a more obscure way than I thought initially.
I tested locally with:
```
last_good_commit=$(curl -L \
            -H "Accept: application/vnd.github+json" \
            -H "X-GitHub-Api-Version: 2022-11-28" \
            "https://api.github.com/repos/vlang/v/actions/workflows/18477644/runs?branch=master&status=success&per_page=10" \
            | jq -r '.workflow_runs[5].head_sha') && echo $last_good_commit
```
and it retrieved `00dd0bf9383c601b3b96a57191f6b76335f44538` , which is a recent commit, but not the latest, which is very good.

Reply:
It did not need the `Authorization: Bearer` header.

Reply:
Is it going up just 3 `^^^` parents better vs always going up a few "good" parents (can also be 3)?

In the first commit I used a simpler version, which also included a comment about the used workflow.
```yml
# Derive commit sha from an older successful fast workflow on master that was able to build V.
# The workflow used below is `Path Testing CI` (18477644).
last_good_commit=$(curl -s "https://api.github.com/repos/vlang/v/actions/workflows/18477644/runs?branch=master&status=success&per_page=1" | jq -r '.workflow_runs[0].head_sha')
echo "last_good_commit=$last_good_commit"
```

> It did not need the Authorization: Bearer header, which is also nice.
Yes it's possible to check the URL without authorization.

Those api endpoints have fairly low rate limits though. So I tried to give the version without the auth token a few consecutive pushes. It started to fail then, and could be resolved by adding the auth token.

Reply:
- Regarding the variable name: maybe using something better than `last_good_commit` as it's not exactly the `last`. `old_good_commit`? 

- Regarding rate limits: Idk exactly under which ip address runners are tracked. It can be aspects outside this repository why they already come with a lower rate limit.


-  For local testing: https://api.github.com/rate_limit
-  Check again after opening the url to get the good commmit: https://api.github.com/repos/vlang/v/actions/workflows/18477644/runs?branch=master&status=success&per_page=1

Authorized, the rate limit is 5000 instead of 60.

Reply:
> In the first commit I used a simpler version, which also included a comment about the used workflow.

Please restore that comment. In several weeks/months, it would be hard again to re-acquire that knowledge, and it may be needed, in the worst possible time for little side quests.

Reply:
Thanks for the constructive review.

If there are further unclear points lets address them.

Reply:
No, I think it is good to merge as it is now.